platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.kt

```diff
@@ -652,22 +666,26 @@ open class MapSnapshotter(context: Context, options: Options) {
             fullyLoaded = true
             val builder = options.builder
             if (builder != null) {
-                for (source in builder.sources) {
-                    nativeAddSource(source, source.nativePtr)
-                }
-                for (layerWrapper in builder.layers) {
-                    if (layerWrapper is LayerAtWrapper) {
-                        addLayerAt(layerWrapper.getLayer(), layerWrapper.index)
-                    } else if (layerWrapper is LayerAboveWrapper) {
-                        addLayerAbove(layerWrapper.getLayer(), layerWrapper.aboveLayer)
-                    } else if (layerWrapper is LayerBelowWrapper) {
-                        addLayerBelow(layerWrapper.getLayer(), layerWrapper.belowLayer)
-                    } else {
-                        addLayerBelow(layerWrapper.layer, MapboxConstants.LAYER_ID_ANNOTATIONS)
+                try {
+                    for (source in builder.sources) {
+                        nativeAddSource(source, source.nativePtr)
                     }
-                }
-                for (image in builder.images) {
-                    addImage(image.id, image.bitmap, image.isSdf)
+                    for (layerWrapper in builder.layers) {
+                        if (layerWrapper is LayerAtWrapper) {
+                            addLayerAt(layerWrapper.getLayer(), layerWrapper.index)
+                        } else if (layerWrapper is LayerAboveWrapper) {
+                            addLayerAbove(layerWrapper.getLayer(), layerWrapper.aboveLayer)
+                        } else if (layerWrapper is LayerBelowWrapper) {
+                            addLayerBelow(layerWrapper.getLayer(), layerWrapper.belowLayer)
+                        } else {
+                            addLayerBelow(layerWrapper.layer, MapboxConstants.LAYER_ID_ANNOTATIONS)
+                        }
+                    }
+                    for (image in builder.images) {
+                        addImage(image.id, image.bitmap, image.isSdf)
+                    }
+                } catch (e: IllegalStateException) {
+                    e.printStackTrace()
```

Comment:
Did you left this in on purpose?

Reply:
Yes. Just fixing the crash not the cause. Once the cause is fixed we can reuse fully previous code.